### PR TITLE
Give random seed in config file

### PR DESCRIPTION
**Pull request by markusdregi**
**Task**
_Functionality for providing the random seed directly in the configuration file._

**Approach**
- Support keyword `RANDOM_SEED` in configuration file and configuration dict.
- If both `RANDOM_SEED` and `LOAD_SEED` is provided, `RANDOM_SEED` takes precedence.
- The `RANDOM_SEED` is written to the log when `enkf_main` is instantiated, so that one can replicate the current results if wanted.

**Remarks**
Do we want to deprecate `LOAD_SEED` and `STORE_SEED` or should both possibilities live side by side? Both are feasible solutions. The later is the current status, but deprecation is only a log-message away..

Note that most of the diff is new tests. 

**Pre un-WIP checklist**
- [x] Statoil tests pass locally
- [ ] Have completed graphical integration test steps

### CODE REVIEW

**Comment by joakim-hove** [comment]
I will look into this later today.

**Comment by joakim-hove** [comment]
> Do we want to deprecate LOAD_SEED and STORE_SEED or should both possibilities live side by side? Both are feasible solutions. The later is the current status, but deprecation is only a log-message away..

My personal wish is to take it even further: just remove the `LOAD_SEED` and `STORE_SEED` functionality altogether. This is rarely used expert functionality, the new functionality is equivalent - if not identical, the gain in reduced complexity is just too tempting. In addition to actually removing the load/store functionality from the code you can add a message to the config system [here](https://github.com/Statoil/libres/blob/master/libenkf/src/rng_config.c#L129):
```
config_install_message(parser, CONFIG_LOAD_SEED_KEY, "The LOAD_SEED functionality has been removed - use the alternative keyword `RANDOM_SEED'");
```

File: libenkf/src/rng_config.c
Comment (joakim-hove) [inline libenkf/src/rng_config.c]: The `util_safe_free( )` originates in a misunderstanding - it is safe to call `free( NULL )`.

File: libenkf/src/rng_config.c
Comment (joakim-hove) [inline libenkf/src/rng_config.c]: The lines 152-159 could be replaced with a call: `rng_config_log_seed( rng_config )`- which again unconditionally called the internal rng to get state; should not be necessary to distinguish between seeds explicitly provided by the user and default initialized seeds?? If this is indeed possible we can also remove the `rng_manager_get_state()` function completely.

File: libenkf/src/rng_config.c
Comment (joakim-hove) [inline libenkf/src/rng_config.c]: These two lines can be collapsed to: 
```
rng_config->random_seed = util_realloc_string_copy( rng_config->random_seed, random_seed );
```

File: libenkf/src/rng_manager.c
Comment (joakim-hove) [inline libenkf/src/rng_manager.c]: Maybe `rng_manager_export_state( )` is better - since the function actually is `void`! But see other comment in `rng_config.c` - unless I have misunderstood something else I would actually recommend the complete removal of this function.

**Comment by joakim-hove** [comment]
The `rng_manager.c` implementation has a couple of magic '4' values sprinkled around; I am *not* suggesting we should go fully general and ask the rng for state size, but could at least introduce a slightly documented symbol:
```
#define RNG_STATE_SIZE 4
```

**Review by joakim-hove — CHANGES_REQUESTED** [review]
This all over very good - but I would prefer if the `LOAD_SEED` and `STORE_SEED` functionality was completely removed.

**Comment by markusdregi** [comment]
@joakim-hove The reason I kept both `RANDOM_SEED` and `LOAD_SEED` around is that wouldn't the immediate removal of a key in the configuration file impose a bump of the major version if staying true to semver?

If we want the `LOAD_SEED` and `STORE_SEED` functionality gone we can either just remove it completely as you suggest or we can deprecate it (which is technically very easy now) and remove it next time we do an appropriate change of the version. I'll let you decide and will update my code accordingly.

File: libenkf/src/rng_config.c
Comment (markusdregi) [reply libenkf/src/rng_config.c]: The thing is that the journey from a random seed in the configuration file to the internal state of the `rng_manager` is not injective. Due to this, if specifying `RANDOM_SEED 1234` in the configuration file, you might very well be asked to add `123412341234123412341234123412341234123` to the configuration file to repeat the experiment, which is kind of odd and unexpected.

Comment (joakim-hove) [reply libenkf/src/rng_config.c]: >Due to this, if specifying RANDOM_SEED 1234 in the configuration file, you might very well be asked to add 123412341234123412341234123412341234123 to the configuration file to repeat the experiment, which is kind of odd and unexpected.

Well - we can live with that - in my opinon.

**Comment by joakim-hove** [comment]
> If we want the LOAD_SEED and STORE_SEED functionality gone we can either just remove it completely as you suggest or we can deprecate it (which is technically very easy now) and remove it next time we do an appropriate change of the version. I'll let you decide and will update my code accordingly.

OK - I think it should be removed in the long run; so then I suggest deprecation now.

**Review by joakim-hove — APPROVED** [review]

**Comment by joakim-hove** [comment]
Resolve comments as you see fit - and this should be ready to merged.

File: libenkf/src/rng_config.c
Comment (markusdregi) [reply libenkf/src/rng_config.c]: Aha, I never really understood the reason for `util_safe_free` to exist. If you agree, we could deprecate it then?

**Comment by markusdregi** [comment]
Addressed all comments. Will merge when Travis and Jenkins are green.

**Pull request merged**
